lynx/common/actions/chop.py

```diff
@@ -1,30 +1,41 @@
 from dataclasses import dataclass
 
 from lynx.common.actions.action import Action
-from lynx.common.vector import Vector
-from lynx.common.actions.remove_object import RemoveObject
+from lynx.common.actions.common_requirements import CommonRequirements
 from lynx.common.actions.create_object import CreateObject
+from lynx.common.actions.remove_object import RemoveObject
+from lynx.common.enums import Direction
 from lynx.common.object import Object
-from lynx.common.actions.common_requirements import CommonRequirements
+from lynx.common.vector import Vector
+
 
 @dataclass
 class Chop(Action):
-	"""
-	Simple action used to hit/destroy object, which stands on the given destination.
-	"""
-	object_id: int = -1
-	target_position: Vector = Vector(1, 0)
+    """
+    Simple action used to hit/destroy object, which stands on the given destination.
+    """
+    object_id: int = -1
+    # target_position: Vector = Vector(1, 0) # Currently we want to chop the tree in specific direction
+    direction: Vector = Direction.NORTH.value
+
+    def apply(self, scene: 'Scene') -> None:
+        target_position = self._get_target_position(scene)
+        objects_on_square = scene.get_objects_by_position(target_position)
+        # TODO put 'Tree' into an Enum
+        chopped_object: Object = \
+        list(filter(lambda object_on_square: object_on_square.name == 'Tree', objects_on_square))[0]
+        if chopped_object:
+            remove_action = RemoveObject(chopped_object.id)
+            log = Object(id=scene.generate_id(), name='Log', pushable=True, walkable=False,
+                         position=chopped_object.position,
+                         pickable=True)
+            create_action = CreateObject(log.serialize())
+            scene.add_to_pending_actions(remove_action.serialize())
+            scene.add_to_pending_actions(create_action.serialize())
 
-	def apply(self, scene: 'Scene') -> None:
-		objects_on_square = scene.get_objects_by_position(self.target_position)
-		for object in objects_on_square:
-			# TODO: put all object names into an Enum
-			if object.name == 'Tree':
-				remove_action = RemoveObject(object.id)
-				log = Object(id=scene.generate_id(), name='Log', position=object.position, pickable=True)
-				create_action = CreateObject(log.serialize())
-				scene.add_to_pending_actions(remove_action.serialize())
-				scene.add_to_pending_actions(create_action.serialize())
+    def satisfies_requirements(self, scene: 'Scene') -> bool:
+        return CommonRequirements.is_in_range(scene, self.object_id, self._get_target_position(scene), 1) \

```

Comment:
Do we need this requirement, if we use Directions?

Reply:
Generally, if we want, we can also pass any vector to chop(), so I think it's necessary.